### PR TITLE
fix: only fallback to OCI image when current media type is OCI artifact

### LINKED ISSUE #672: Check media type of manifest before fallback to OCI image for `oras push` and `oras atatch`
Curious why we chose names that result in double-negatives?  For example *"is it not not-supported"* vs *"is it supported"*.

_Originally posted by @nima in https://github.com/oras-project/oras/pull/665#discussion_r1013118610_

### PR DESCRIPTION
**Pull request by qweeah**
Resolves #672 

Signed-off-by: Billy Zha <jinzha1@microsoft.com>

### CODE REVIEW

File: cmd/oras/push.go [hunk 1]
```diff
@@ -276,22 +276,25 @@ func pushArtifact(dst *remote.Repository, pack packFunc, packOpts *oras.PackOpti
 	return root, nil
 }
 
-func isArtifactUnsupported(err error) bool {
+func noFallbackToOciImage(root ocispec.Descriptor, err error) bool {
+	if root.MediaType != ocispec.MediaTypeArtifactManifest {
+		return true
+	}
```
Comment (shizhMSFT) [inline cmd/oras/push.go:282]: Nice catch

File: cmd/oras/push.go [hunk 2]
```diff
@@ -276,22 +276,25 @@ func pushArtifact(dst *remote.Repository, pack packFunc, packOpts *oras.PackOpti
 	return root, nil
 }
 
-func isArtifactUnsupported(err error) bool {
+func noFallbackToOciImage(root ocispec.Descriptor, err error) bool {
```
Comment (shizhMSFT) [inline cmd/oras/push.go:279]: Making `isArtifactUnsupported` to `noFallbackToOciImage` might improve the readability of the invokers but make the readability of the implementation itself worse and easier to have bugs in it.

File: cmd/oras/push.go [hunk 3]
```diff
@@ -276,22 +276,25 @@ func pushArtifact(dst *remote.Repository, pack packFunc, packOpts *oras.PackOpti
 	return root, nil
 }
 
-func isArtifactUnsupported(err error) bool {
+func skipFallbackToImageManifest(root ocispec.Descriptor, err error) bool {
```
Comment (Wwwsylvia) [inline cmd/oras/push.go:279]: Can we have a documentation for this function?

**Review by shizhMSFT — APPROVED** [review]
LGTM

File: cmd/oras/push.go (hunk 3 above)
Comment (qweeah) [reply cmd/oras/push.go:279]: Do we still need it now after the refactoring?

File: cmd/oras/push.go (hunk 2 above)
Comment (qweeah) [reply cmd/oras/push.go]: Do you still have this concern after the refactoring?

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Pull request merged**